Code/Tools/SceneAPI/SceneUI/RowWidgets/NodeTreeSelectionWidget.cpp

```diff
@@ -208,54 +208,57 @@ namespace AZ
                 size_t result = 0;
                 AZStd::unique_ptr<DataTypes::ISceneNodeSelectionList> tempList(m_list->Copy());
                 Utilities::SceneGraphSelector::UpdateNodeSelection(graph, *tempList);
-                size_t selectedCount = tempList->GetSelectedNodeCount();
-                for (size_t i = 0; i < selectedCount; ++i)
-                {
-                    Containers::SceneGraph::NodeIndex index = graph.Find(tempList->GetSelectedNode(i));
-                    if (!index.IsValid())
+                tempList->EnumerateSelectedNodes(
+                    [&](const AZStd::string& nodeName)
                     {
-                        continue;
-                    }
-
-                    AZStd::shared_ptr<const DataTypes::IGraphObject> object = graph.GetNodeContent(index);
-                    if (!object)
-                    {
-                        continue;
-                    }
+                        Containers::SceneGraph::NodeIndex index = graph.Find(nodeName);
+                        if (!index.IsValid())
+                        {
+                            return true;
+                        }
 
-                    if (m_filterTypes.empty() && m_filterVirtualTypes.empty())
-                    {
-                        result++;
-                        continue;
-                    }
+                        AZStd::shared_ptr<const DataTypes::IGraphObject> object = graph.GetNodeContent(index);
+                        if (!object)
+                        {
+                            return true;
+                        }
 
-                    bool foundType = false;
-                    for (const Uuid& type : m_filterTypes)
-                    {
-                        if (object->RTTI_IsTypeOf(type))
+                        if (m_filterTypes.empty() && m_filterVirtualTypes.empty())
                         {
                             result++;
-                            foundType = true;
-                            break;
+                            return true;
```

Comment:
don't really see a need for the `result++;` above this line anymore

Reply:
keep in mind the "return true" here is inside of EnumerateSelectedNodes() to tell it to keep processing, and the result value lives outside the EnumerateSelectedNodes() call. Keeping the result++ here should produce equivalent results to what was previously on lines 226-230.